docs/user/api/secrets.md

```diff
@@ -0,0 +1,118 @@
+# Secret reporting API
+
+!!! warning "Not publicly available"
+
+    Note that this API is only available on a case-by-case basis. Please
+    contact admin@pypi.org if you would like to integrate with this API.
+
+Third parties integrate with PyPI to find, identify and revoke API tokens that
+are accidentally made public. The following partners currently report publicly
+exposed API tokens to PyPI:
+
+* https://github.com
+* https://deps.dev
+
+All PyPI users that use API tokens are opted into this by default, and no
+action is necessary to benefit from this.
+
+This API is for third parties who may find PyPI API tokens and wish to
+report them to PyPI.
+
+## Detecting the PyPI secret format
+
+A PyPI API token is a string consisting of a prefix (``pypi``), a separator
+(`-`) and a string representing a Macaroon `base64` serialized with
+[PyMacaroon]:
+
+    pypi-[A-Za-z0-9-_]{85,}
+
+The `base64` string will not be shorter than 85 characters. A token can be
+arbitrarily long because we may add arbitrary caveats to the serialized
+Macaroon.
+
+## Integrating
+
+PyPI has adopted the [GitHub secret scanning reporting pattern].
+
+### Public key identifier & signature
+
+PyPI expects every request to this API to include two headers:
+
+* A header containing a public key identifier
+* A header containing a signature of the raw message body using this key
+
+The names of these headers can be arbitrary and should be provided to PyPI at
+integration time. They will be verified for every request.
+
+### Public key verification
+
+PyPI expects to be able to verify the public key used to sign the request at a
+URL provided at integration time. This URL structure is arbitrary but should
+exist at a trusted domain.
+
+The response from a GET request to this URL should return a JSON document with
+the following example structure:
+
+```json
+{
+  "public_keys": [
+    {
+      "key_identifier": "90a421169f0a406205f1563a953312f0be898d3c7b6c06b681aa86a874555f4a",
+      "key": "-----BEGIN PUBLIC KEY-----\nMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAE9MJJHnMfn2+H4xL4YaPDA4RpJqUq\nkCmRCBnYERxZanmcpzQSXs1X/AljlKkbJ8qpVIW4clayyef9gWhFbNHWAA==\n-----END PUBLIC KEY-----\n",
+      "is_current": false
```

Comment:
Not sure if this is worth mentioning or now, but flagging: the current `_check_signature` helper assumes that the signature is an ECDSA sig, and that the digest is SHA-256. 

As such, integrating parties should be prepared to provide P-256/384/521 keys, and use SHA-256 only (not SHA-384 or SHA-512, despite those being common with P-384 and P-521 respectively).

Ref: https://github.com/pypi/warehouse/blob/7ce8cee7a09c03b0f252e4a0c7da0787ab08bcdb/warehouse/integrations/__init__.py#L122

Reply:
Added in f119687, PTAL.